src/python/pants/option/config.py

```diff
@@ -193,10 +182,10 @@ def _possibly_interpolate_value(
     ) -> str:
         """For any values with %(foo)s, substitute it with the corresponding value from DEFAULT or
         the same section."""
+        # TODO(benjy): I wonder if we can abuse ConfigParser to do this for us?
 
         def format_str(value: str) -> str:
-            # Because dictionaries use the symbols `{}`, we must proactively escape the symbols so
-            # that .format() does not try to improperly interpolate.
+            # Escape embedded { and } characters, so that .format() does not act on them.
```

Comment:
This is necessary regardless of dictionaries. E.g., a string could have curly braces in it for some other reason, so might as well simplify the comment.